core/common/src/BriefcaseTypes.ts

```diff
@@ -166,3 +164,26 @@ export interface UpgradeOptions {
   /** Options that control whether a profile upgrade should be performed when opening a file */
   profile?: ProfileOptions;
 }
+
+/**
+ * The state of the schemas in the Db compared with what the current version of the software expects
+ * Note: The state may vary depending on whether the Db is to be opened ReadOnly or ReadWrite. Generally the
+ * requirements for opening a Db ReadOnly are less stringent than opening it ReadWrite.
+ * @beta
+ */
+export enum SchemaState {
+  /** The schemas in the Db are up-to-date, and do not need to be upgraded before opening it with the current version of the software */
+  UpToDate,
+
+  /** It's required that the schemas in the Db be upgraded before it can be opened with the current version of the software */
+  UpgradeRequired,

```

Comment:
even readonly? Please explain what you can do readonly vs. readwrite.

Reply:
Added a few more comments now - please take a look.